SPEC/PUBSUB.md

```diff
@@ -7,13 +7,13 @@ pubsub API
 
 ##### `Go` **WIP**
 
-##### `JavaScript` - ipfs.pubsub.subscribe(topic, options, handler, callback)
+##### `JavaScript` - ipfs.pubsub.subscribe(options, callback)
 
-- `topic: string`
-- `options: Object` - (Optional), might contain the following properties:
-  - `discover`: type: Boolean - Will use the DHT to find other peers.
-- `handler: (msg) => ()` - Event handler which will be called with a message object everytime one is received. The `msg` has the format `{from: string, seqno: Buffer, data: Buffer, topicIDs: Array<string>}`.
-- `callback: (Error) => ()` (Optional) Called once the subscription is established.
+- `options: Object`: Object containing the following properties:
+  - `topic: string`
+  - `discover: Boolean` - Will use the DHT to find other peers.
+  - `handler: (msg) => ()` - Event handler which will be called with a message object everytime one is received. The `msg` has the format `{from: string, seqno: Buffer, data: Buffer, topicIDs: Array<string>}`.
```

Comment:
The handler moves into the options object? That's a bit of a weird pattern, much nicer when it was separated.

Reply:
I believe your proposal of:

```
ipfs.pubsub.subscribe(topic, handler, options, callback)
ipfs.pubsub.unsubscribe(topic, handler, callback)
```

Pretty much solves the main issue. I like it!